stan/math/prim/fun/hypergeometric_1F0.hpp

```diff
@@ -0,0 +1,40 @@
+#ifndef STAN_MATH_PRIM_FUN_HYPERGEOMETRIC_1F0_HPP
+#define STAN_MATH_PRIM_FUN_HYPERGEOMETRIC_1F0_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/err/check_less.hpp>
+#include <stan/math/prim/fun/boost_policy.hpp>
+#include <boost/math/special_functions/hypergeometric_1F0.hpp>
+#include <cmath>
+
+namespace stan {
+namespace math {
+
+/**
+ * Returns the Hypergeometric 1F0 function applied to the
+ * input arguments:
+ * \f$ _1F_0(a;;z) = \sum_{k=1}^{\infty}\frac{\left(a\right)_kz^k}{k!}\f$
+ *
+ * \f$ \frac{\partial _1F_0\left(a;;z\right)}{\partial a} =
+ *    -\left(1-z\right)^{-a}\log\left(1 - z\right) \f$
+ *
+ * \f$ \frac{\partial _1F_0\left(a;;z\right)}{\partial z} =
+ *    a\left(1-z\right)^{-a-1} \f$
+ *
+ * @tparam Ta Arithmetic type of 'a' argument
+ * @tparam Tz Arithmetic type of 'z' argument
+ * @param[in] a Scalar 'a' argument
+ * @param[in] z Scalar z argument
+ * @return Hypergeometric 1F0 function
+ */
+template <typename Ta, typename Tz, require_all_arithmetic_t<Ta, Tz>* = nullptr>
+auto hypergeometric_1f0(const Ta& a, const Tz& z) {
```

Comment:
Please use proper return type.

Reply:
Given the type assumptions being made, is `Ta` and `Tz` just `double` and `double` only? 

If I'm reading this properly, this isn't vectorized or anything, is it?

Reply:
This is to allow all combinations of `int`/`double` (as well as `size_t`,`float`, etc.)